src/App.jsx

```diff
@@ -26,8 +34,9 @@ function App() {
       <ul>
         {todos.map((todo, index) => (
           <li key={todo}>
-            {todo}
+            {isEditableList[index] ? <input /> : <span>{todo}</span> }
```

Comment:
Now you learned a myriad of concepts, namely conditional rendering, using refs for manipulating DOM nodes and manipulation state in an immutable way which is always the standard.